tests/AbstractLoopTest.php

```diff
@@ -589,11 +589,20 @@ public function testSignalsKeepTheLoopRunningAndRemovingItStopsTheLoop()
         $this->assertRunFasterThan(1.6);
     }
 
+    public function testTimerIntervalBelowZeroRunsImmediately()
+    {
+        $this->loop->addTimer(-1, function () {});
+
+        $this->assertRunFasterThan(0.002);
+    }
```

Comment:
We currently have this odd split in our tests suite, can you move this to the similar tests in `AbstractTimerTests`?

Other than that, this test *looks* good to me, but we've seen similar tests fail sporadically due to timer inaccuracies, see also https://github.com/reactphp/event-loop/blob/babf91e65eea396681a8abda8ba91589a21da63b/tests/Timer/AbstractTimerTest.php#L28-L45

Perhaps you can update this to use a similar logic to make sure an inaccurate platform does not cause this test to fail?

Reply:
Moved.